### DISCUSSION TITLE
What are Dropout Layers ?

### DISCUSSION THREAD

**Discussion by trustunsafe** [gauravreddy08/food-vision#2] [Q&A] [answered]
Hey Gaurav,
I was trying to replicate your idea of using EfficientNetB1 but i got validation accuracy of about 72%....
Then i looked at your model training notebook and found a layer that i was not using...(Dropout)
Can you explain me how this layer is helping out in the training?
What does it do?

**Comment by gauravreddy08** [comment, marked as answer]
Hey @MEGA-BLASTER2004 

Dropout layer drops out few neurones in the network. This to reduces the problem of overfitting. Here, a pic for clear understanding :

![image](https://user-images.githubusercontent.com/57211163/121685419-89001800-cadd-11eb-85ce-60c6ca457b79.png)

When you use a complex model like **`EfficientNetB1`** the model tends to overfit on the data. To prevent overfitting we need to reduce the complexity, by dropping out few neurones. And that's what Dropout Layer comes into the play. 

If you still can't get over it check this out : **[Deep Learning: Using Dropout Layers in CNNs to prevent Overfitting!]( https://www.youtube.com/watch?v=J-oobN-OXM0)**

> I feel like even without dropout layers you can get accuracy over 72% (i may be wrong). Check if you set the model layer's unfrozen. Cause Daniel usually does model training in 2 parts : 
> * Feature Extraction 
> * Fine Tuning

> Where as in my case, I got good results directly by fine-tuning the model

----

> **If you're wondering what overfitting is : The model trains so well `train data` that get confused on `test data`. Here's a cool example :**
> 
> Let's say you go to a new country and a cab driver takes advantage of you, as you're new to the place.
> He charges you more than usual.
> We humans, immediately come to a conclusion that all the cab drivers are a*holes. 
> Thats what model does, it trains on `train data` so well that it generalises and makes no improvement on `test data`.

**Reply by gauravreddy08** [reply]
I hope this answers your question 😄 

> And it's great to see you trying to replicate what I did. It means a lot. Good Luck. 👍🏻

**Reply by trustunsafe** [reply]
Yeah!
Thanks...I had frozen the base model layers...😅

**Reply by gauravreddy08** [reply]
Hey, I see that you're also learning Deep Learning. I'd love to collaborate or just talk about ML/DL. 

If you're down, hit me up. 

> You can connect with me through the links on my profile itself.

**Reply by trustunsafe** [reply]
Sure!
